restli-docgen/src/main/java/com/linkedin/restli/docgen/DefaultDocumentationRequestHandler.java

```diff
@@ -99,11 +98,23 @@ private void initializeRenderers()
     final ResourceSchemaCollection resourceSchemas = ResourceSchemaCollection.loadOrCreateResourceSchema(_rootResources);
     final RestLiResourceRelationship relationships = new RestLiResourceRelationship(resourceSchemas, schemaResolver);
 
-    _htmlRenderer = new RestLiHTMLDocumentationRenderer(_config.getServerNodeUri(),
+    _htmlRenderer = getHtmlDocumentationRenderer(schemaResolver, relationships);
+    _jsonRenderer = getJsonDocumentationRenderer(schemaResolver, relationships);
+  }
+
+  protected RestLiDocumentationRenderer getHtmlDocumentationRenderer(DataSchemaResolver schemaResolver,
+      RestLiResourceRelationship relationships)
+  {
+    return new RestLiHTMLDocumentationRenderer(_config.getServerNodeUri(),
         relationships,
         new VelocityTemplatingEngine(),
         schemaResolver);
-    _jsonRenderer = new RestLiJSONDocumentationRenderer(relationships);
+  }
+
+  protected RestLiDocumentationRenderer getJsonDocumentationRenderer(DataSchemaResolver schemaResolver,
```

Comment:
If I understand correctly, this (and `#renderDataModel`) is the "hook", right? Users will extend this class and override this method, using this subclass as the documentation handler instead of the default one. Please correct me if I'm wrong.

Reply:
you are correct. Users can also customize other render* methods.